huff_utils/src/files.rs

```diff
@@ -35,6 +43,123 @@ impl<'a> FullFileSource<'a> {
+    pub base_dir: String,
+}
+
+impl Remapper {
+    /// Extracts remappings from configuration files.
+    ///
+    /// Currently only supports `foundry.toml` remapping definitions.
+    pub fn new(root: impl AsRef<str>) -> Self {
+        let mut inner = HashMap::<String, String>::new();
+
+        // Gracefully parse remappings from foundry.toml
+        Remapper::from_foundry(root.as_ref(), &mut inner);
+
+        // Return the constructed remappings
+        Self { remappings: inner, base_dir: root.as_ref().to_string() }
+    }
+
+    /// Helper to break apart a remapping gracefully
+    pub fn split(remapping: &str) -> Option<(String, String)> {
+        let mut split = remapping.splitn(2, '=');
+        match split.next() {
+            Some(from) => split.next().map(|to| (from.to_string(), to.to_string())),
+            None => None,
+        }
+    }
+
+    /// Parse foundry toml remappings
+    pub fn from_foundry(root: &str, inner: &mut HashMap<String, String>) {
+        // Look for a `foundry.toml` file in the current directory.
+        let path = Path::new(root).join("foundry.toml");
+
+        match fs::File::open(&path) {
+            Ok(f) => {
+                // Open the buffered reader and read foundry.toml
+                let mut data = String::new();
+                let mut br = BufReader::new(f);
+                match br.read_to_string(&mut data) {
+                    Ok(_) => {
+                        tracing::debug!(target: "parser", "Read foundry.toml contents {}", data);
+                        // Parse the foundry.toml file
+                        match data.parse::<toml::Value>() {
+                            Ok(toml) => {
+                                // Extract the remappings
+                                if let Some(toml_table) = toml.as_table() {
+                                    toml_table.iter().for_each(|p| {
+                                        if let Some(profiles) = p.1.as_table() {
+                                            profiles.iter().for_each(|table| {
+                                                if let Some(inner_table) = table.1.as_table() {
+                                                    inner_table.iter().for_each(|_| {
+                                                        if let Some(remapping) = inner_table.get("remappings") {
+                                                            match remapping.as_array() {
+                                                                Some(parsed_remappings) => {
+                                                                    parsed_remappings.iter().for_each(|remapping| {
+                                                                        match remapping.as_str() {
+                                                                            Some(str) => match Remapper::split(str) {
+                                                                                Some((from, to)) => {
```

Comment:
This is beautiful :laughing:

Definitely fulfills its purpose though- any reason not to have a `try_from_foundry` function (or something to that effect) to clean this up?